tezos/api/src/ffi.rs

```diff
@@ -179,6 +179,21 @@ pub struct BeginApplicationResponse {
     pub result: String,
 }
 
+#[cfg_attr(feature = "fuzzing", derive(fuzzcheck::DefaultMutator))]
+#[derive(Error, Serialize, Deserialize, Debug, Clone)]
+pub enum GetLastContextHashesError {
+    #[error("Failed to get the latest context hashes: {message}!")]
+    FailedToGetCurrentHead { message: String },
```

Comment:
In general it's better if we don't use `String` for errors and instead use concrete types, but I see `TezosErrorTrace` contains string errors as well, so we can refactor this later as a part of a refactor to get rid of strings in the error types.

Reply:
The traces that we get from the protocol runner are strings containing the JSON representation of the OCaml error trace, we usually don't do anything with those other than displaying them for debugging purposes.

In the cases where it is needed we can check the "id" field of the json object to figure out what the error is and map it to something concrete (like here https://github.com/tezedge/tezedge/blob/97b3faa5b9ceee56a41bb03554fafba8c623eca0/tezos/api/src/ffi.rs#L532-L558)